scripts/babel-relay-plugin/lib/RelayQLAST.js

```diff
@@ -147,19 +147,14 @@ var RelayQLFragment = function (_RelayQLDefinition) {
 
     _this4.hasStaticFragmentID = isStaticFragment;
     _this4.parentType = parentType;
-    _this4.staticFragmentID = null;
     return _this4;
   }
 
   RelayQLFragment.prototype.getStaticFragmentID = function getStaticFragmentID() {
-    if (this.hasStaticFragmentID && this.staticFragmentID == null) {
-      var suffix = this.context.generateID();
-      // The fragmentLocationID is the same for all inline/nested fragments
-      // within each Relay.QL tagged template expression; the auto-incrementing
-      // suffix distinguishes these fragments from each other.
-      this.staticFragmentID = this.context.fragmentLocationID + ':' + suffix;
+    if (this.hasStaticFragmentID) {
+      return this.getName();
```

Comment:
We should still fall back onto something that is more unique than the automatically generated fragment name.

What about changing the automatically generated fragment name to be `documentName + [index of fragment in document]`?